src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -426,40 +425,38 @@ public void ProduceEnd(Exception ex)
             var hasConnection = false;
             var hasTransferEncoding = false;
             var hasContentLength = false;
-            if (headers != null)
+
+            foreach (var header in _responseHeaders)
```

Comment:
Using _responseHeaders here is an issue because we set the ResponseHeaders property in [the ProduceEnd method on line 391](https://github.com/aspnet/KestrelHttpServer/pull/210/files#diff-3ad18666c41931c02b7bd1f3226cb709R391).

I  think this PR can wait until @lodejard's [flow control PR](https://github.com/aspnet/KestrelHttpServer/pull/207) gets merged which [changes this](https://github.com/aspnet/KestrelHttpServer/pull/207/files#diff-3ad18666c41931c02b7bd1f3226cb709R445).